device/core/devdoc/device_twin_requirements.md

```diff
@@ -3,146 +3,75 @@
 
+**SRS_NODE_DEVICE_TWIN_16_008: [** If the callback passed to the transport is called with an error, the `callback` argument of the `update` method shall be called with that error. **]**
 
-### properties.reported.update method
-`update` is a method which application developers use to send reported state to the service.
+**SRS_NODE_DEVICE_TWIN_18_031: [** If the callback passed to the transport is called with no error, the  `properties.reported.update` shall merge the contents of the patch object into `properties.reported` **]**
 
-**SRS_NODE_DEVICE_TWIN_18_025: [** `properties.reported.update` shall use _sendTwinRequest to send the patch object to the service. **]** 
+**SRS_NODE_DEVICE_TWIN_18_032: [** When merging the patch, if any properties are set to `null`, `properties.reported.update` shall delete that property from `properties.reported`. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_026: [** When calling `_sendTwinRequest`, `properties.reported.update` shall pass `method`='PATCH', `resource`='/properties/reported/', `properties`={}, and `body`=the `body` parameter passed in to `reportState` as a string. **]**   
+**SRS_NODE_DEVICE_TWIN_16_009: [** Once the properties have been merged the `callback` argument of the `update` method shall be called with no argument. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_027: [** `properties.reported.update` shall call `done` with the results from `_sendTwinRequest` **]** 
+### on('properties.desired[.path])
 
-**SRS_NODE_DEVICE_TWIN_18_031: [** `properties.reported.update` shall merge the contents of the patch object into `properties.reported` **]**
+**SRS_NODE_DEVICE_TWIN_16_010: [** When a listener is added for an event name starting with `properties.desired` the `enableTwinDesiredPropertiesUpdates` method of the `Transport` object shall be called with a callback function accepting an optional error argument. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_032: [** When merging the patch, if any properties are set to `null`, `properties.reported.update` shall delete that property from `properties.reported`. **]**
+**SRS_NODE_DEVICE_TWIN_16_011: [** If the callback passed to the transport is called with an error, that error shall be emitted by the Twin object. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_033: [** If `_sendTwinRequest` fails, `properties.reported.update` shall not merge the contents of the patch object into `properties.reported` **]**
+**SRS_NODE_DEVICE_TWIN_18_045: [** If a property is already set when a handler is added for that property, the `Twin` object shall fire a property changed event for the property. **]**
 
+**SRS_NODE_DEVICE_TWIN_16_012: [** When a `twinDesiredPropertiesUpdates` event is emitted by the transport, the property patch passed as argument to the event handler shall be merged with the current desired properties. **]**
```

Comment:
>twinDesiredPropertiesUpdates [](start = 43, length = 28)

The differences between this event and the one the constructor subscribed to?

Reply:
it's the same event. this requirement is about what to do with it, the constructor requirement is about subscribing to it

---
In reply to: [171910926](https://github.com/pierreca/azure-iot-sdk-node/pull/2#discussion_r171910926) [](ancestors = 171910926)